peps/pep-0750.rst

```diff
@@ -1,889 +1,1143 @@
-become a "tag string": an f-string with a "tag function." The tag function is
-a callable which is given a sequence of arguments for the parsed tokens in
-the string.
+More broadly, the inability to transform interpolated values before they are
+combined into a final string limits the utility of f-strings in more complex
+string processing tasks.
 
-Here's a very simple example. Imagine we want a certain kind of string with
-some custom business policies: uppercase the value and add an exclamation point.
+Template strings address these problems by providing 
+developers with access to the string and its interpolated values. 
 
-Let's start with a tag string which simply returns a static greeting:
+For example, imagine we want to generate some HTML. Using template strings, 
+we can define an ``html()`` function that allows us to automatically sanitize 
+content:
 
 .. code-block:: python
 
-    def greet(*args):
-        """Give a static greeting."""
-        return "Hello!"
+    evil = "<script>alert('evil')</script>"
+    template = t"<p>{evil}</p>"
+    assert html(template) == "<p>&lt;script&gt;alert('evil')&lt;/script&gt;</p>"
 
-    assert greet"Hi" == "Hello!"  # Use the custom "tag" on the string
+Likewise, our hypothetical ``html()`` function can make it easy for developers
+to add attributes to HTML elements using a dictionary:
+
+.. code-block:: python
 
-As you can see, ``greet`` is just a callable, in the place that the ``f``
-prefix would go. Let's look at the args:
+    attributes = {"src": "shrubbery.jpg", "alt": "looks nice"}
+    template = t"<img {attributes} />"
+    assert html(template) == '<img src="shrubbery.jpg" alt="looks nice" />'
+
+Neither of these examples is possible with f-strings. By providing a 
+mechanism to intercept and transform interpolated values, template strings
+enable a wide range of string processing use cases.
+
+
+Specification
+=============
+
+Template String Literals
+------------------------
+
+This PEP introduces a new string prefix, ``t``, to define template string literals.
+These literals resolve to a new type, ``types.Template``.
+
+The following code creates a ``Template`` instance:
 
 .. code-block:: python
 
-    def greet(*args):
-        """Uppercase and add exclamation."""
-        salutation = args[0].upper()
-        return f"{salutation}!"
+    from types import Template
+    template = t"This is a template string."
+    assert isinstance(template, Template)
+
+Template string literals support the full syntax of :pep:`701`. This includes
+the ability to nest template strings within interpolations, as well as the ability
+to use all valid quote marks (``'``, ``"``, ``'''``, and ``"""``). Like other string 
+prefixes, the ``t`` prefix must immediately precede the quote. Like f-strings, 
+both lowercase ``t`` and uppercase ``T`` prefixes are supported. Like 
+f-strings, t-strings may not be combined with the ``b`` or ``u`` prefixes. 
```

Comment:
Quick question: t-strings cannot be combined with `b`, `u` or `f`?

Reply:
Thanks @willingc! That's correct, t-strings cannot be combined with neither one of these. Will fix.

Reply:
Yes, very good catch. Thanks!